client/blocks/signup-form/index.jsx

```diff
@@ -391,6 +432,10 @@ class SignupForm extends Component {
 			username: formState.getFieldValue( this.state.form, 'username' ),
 			password: formState.getFieldValue( this.state.form, 'password' ),
 			email: formState.getFieldValue( this.state.form, 'email' ),
+			extra: {
+				first_name: formState.getFieldValue( this.state.form, 'firstName' ),
```

Comment:
You don't have `user_name`, so maybe just use `firstname` and `lastname` here.

Reply:
This is the data that's passed as it to the API request which expects snake_case keys.  
Doing this automagically higher up might be confusing unless we also put in the effort to update some of the wrapper components that add properties manually.